scripts/compile/jsc.sh

```diff
@@ -19,7 +19,7 @@ $PLATFORM_CFLAGS \
 CMAKE_LD_FLAGS=" \
 -latomic \
 -lm \
--lc++_shared \
+-static-libstdc++ \
```

Comment:
> upgrade NDK to 23.2.8568313. we need newer versions because the newer webkit needs std::filesystem that doesn't support on NDK r21. also, because the NDK r23 libc++_shared.so has std::filesystem symbols that older version doesn't have. to prevent developers use pickFirst to pick wrong libc++_shared.so, i turned to use link static c++ runtime instead.

left it on pr description. feel free to let me know if that's not clear.

Reply:
This does not sound like a safe change. e.g. if we have jsc with statically linked stl and e.g. expo-gl or react-native build with shared stl then we are including stl twice and potentially different versions of it

https://developer.android.com/ndk/guides/cpp-support#sr

> In this situation, the STL, including and global data and static constructors, will be present in both libraries. The runtime behavior of this application is undefined, and in practice crashes are very common. Other possible issues include:
>
> Memory allocated in one library, and freed in the other, causing memory leakage or heap corruption.
Exceptions raised in libfoo.so going uncaught in libbar.so, causing your app to crash.
Buffering of std::cout not working properly.


I just wanted to mention my concern on that, I have only a minimal idea of how all this works, so if you think it's a safe change I'm fine with it

Reply:
interesting! i would say thankfully the libjsc.so is really independency. the exceptions, memory allocations are not across other libraries. the JSCRuntime.cpp implementation in react-native only relies on C interfaces, so it might be safer in this case.

otherwise, TBH i don't have a better solution toward the inconsistent libc++_shared.so issues.